flexmeasures/ui/templates/crud/account.html

```diff
@@ -1,183 +1,394 @@
+                <td class="d-none">/users/{{ user.id }}</td>
+              </tr>
+              {% endfor %}
+            </tbody>
+          </table>
+        </div>
+      </div>
+      <div class="card">
+        <h3>Assets</h3>
+        <div class="table-responsive">
+          <table
+            class="table table-striped table-responsive paginate nav-on-click"
+            title="View this asset"
+          >
+            <thead>
+              <tr>
+                <th><i class="left-icon">Name</i></th>
+                <th>Location</th>
+                <th>Asset ID</th>
+                <th>Account</th>
+                <th>Sensors</th>
+                <th class="no-sort">Status</th>
+                <th class="text-right no-sort">
+                  {% if user_can_create_assets %}
+                  <form action="/assets/new" method="get">
+                    <button
+                      class="btn btn-sm btn-responsive btn-success create-button"
+                      type="submit"
+                    >
+                      Create new asset
+                    </button>
+                  </form>
+                  {% endif %}
+                </th>
+                <th class="d-none">URL</th>
+              </tr>
+            </thead>
+            <tbody>
+              {% for asset in assets %}
+              <tr>
+                <td>
+                  <i
+                    class="{{ asset.generic_asset_type.name | asset_icon }} left-icon"
+                    >{{ asset.name }}</i
+                  >
+                </td>
+                <td>
+                  {% if asset.latitude and asset.longitude %} LAT: {{
+                  "{:,.4f}".format( asset.latitude ) }} LONG: {{
+                  "{:,.4f}".format( asset.longitude ) }} {% endif %}
+                </td>
+                <td>{{ asset.id }}</td>
+                <td>
+                  {% if asset.owner %} {{ asset.owner.name }} {% else %} PUBLIC
+                  {% endif %}
+                </td>
+                <td>{{ asset.sensors | length }}</td>
+                <td>
+                  <a href="/assets/{{ asset.id }}/status">
+                    <button type="button" class="btn">Status</button>
+                  </a>
+                </td>
+                <td class="text-right"></td>
+                <td class="d-none">/assets/{{ asset.id }}</td>
+              </tr>
+              {% endfor %}
+            </tbody>
+          </table>
+        </div>
+      </div>
+    </div>
+    <div class="col-md-2"></div>
+  </div>
 </div>
-<div class="col-md-2"></div>
 {% block paginate_tables_script %} {{ super() }} {% endblock %}
+<script type="text/javascript">
+  const basePath = window.location.origin;
+  const form = document.getElementById("editaccount");
+  const toastElement = document.getElementById("toastcontainer");
+  const toastDescription = document.getElementById("toastbody");
+
+  function showToast(message, type) {
```

Comment:
I think adding Toast is a good innovation that FlexMeasures should have. I believe we only have defined a way to show [messages from the server](https://github.com/FlexMeasures/flexmeasures/blob/main/flexmeasures/ui/templates/base.html#L167).

But then we should introduce it in a way that lets all pages make use of it. Like it is now, inly the account page can use it.

[We actually are using Toast in the smartbuildings plugin](https://github.com/SeitaBV/smart-buildings/blob/main/flexmeasures_smartbuildings/ui/templates/base_flexmeasures_smartbuildings.html#L45). Once we support Toast in FlexMeasures itself, maybe we can stop including extra js libraries there. I believe [smartbuildings imports toastify CSS and JS library](https://github.com/SeitaBV/smart-buildings/blob/main/flexmeasures_smartbuildings/ui/templates/base_flexmeasures_smartbuildings.html#L12) because it was initially written for Bootstrap3 and we moved to Boostrap5 recently. But I am guessing.

So let's move `toastcontainer` to a central place in the base template (maybe right where we keep the server's `msg` element). Also move `showToast()` to a central place (ui/static/js/flexmeasures.js).

Also, I like the examples on [the Boostrap documentation](https://getbootstrap.com/docs/5.0/components/toasts/#live). The way they appear with `position-fixed` in the bottom right.

Reply:
I will implement this after approval on the current implementation before applying it globally.

Reply:
Okay, that's a way to do it, then we'll have one round more of reviewing but I can keep the next iteration to be just about that, nothing else. (not much time before the weekend...)

Reply:
Let me know when the current implementation is done in this PR. That's your call. Is it already?

Reply:
yes it is, I forgot to hit re-request